### DISCUSSION TITLE
Pausing a track reset queue back to where it originally started from

### DISCUSSION THREAD

**Discussion by PureTryOut** [music-assistant/support#3024] [Q&A] [answered]
If I start playing a certain queue and let it play, say, 5 tracks and then pause and resume it, the queue will go back to the very first song. If I manually skip a few songs, say to track 7, it then uses that point as the place to reset to the next time I press pause.

This basically means if I've had a session of listening to a playlist, the next time it'll just do those same songs again and I have to manually skip to where I stopped last time. This becomes very annoying quickly :sweat_smile: 

The setup is Music Assistant 2.2.7 (although I tested it as well with 2.3.0b29) as it's own Podman container (which was a hassle to connect to my dockerized/Podman home-assistant, but that's an issue on it's own) playing as a Snapcast server with a Snapclient on a different machine. The source is Spotify, everything works otherwise.

I did find https://github.com/music-assistant/hass-music-assistant/issues/2893 but it's about Apple Music rather than Spotify and someone got told off for bringing Spotify into it (even though the problem seems to be the exact same) so I'm not sure it's related.

**Comment by OzGav** [comment]
https://music-assistant.io/player-support/snapcast/ see the fifth last note

**Reply by PureTryOut** [reply]
Ok, but that's weird. I also use Snapcast through a piece of software called Mopidy  (a comparable music server but no home-assistant integration) and that has no such issues. It also uses Librespot for it's Spotify integration. Why would Snapcast limitations prevent proper pause functionality here while it doesn't for Mopidy?

**Reply by OzGav** [reply]
Mopidy is not MA. They have their own requirements. @SantiagoSotoC may be able to expand further

**Reply by SantiagoSotoC** [reply]
The problem is that snapcast does not report Timestamps in the control api, only to the clients. And MA requires it for pausing, plus there is the Buffer which makes the idea of pausing not very accurate

**Reply by PureTryOut** [reply]
Interesting, thanks for the information! I can't find an issue about this on the Snapcast repo, has this been reported upstream so someone could take a shot at adding this missing information to the API?

**Reply by SantiagoSotoC** [reply]
I think that more than a problem it would be a new functionality, but I think that if it is going to be requested @marcelveldt , should give us a hand to write it because I do not know well what is necessary, he has more experience in how other systems like sonos among others work.

**Reply by marcelveldt** [reply]
Snapcast does not support stream pausing but that doesn't mean you cant pause in MA. I really do not understand where all that unclarity comes from. If a player does not support native pause (and snapcast clearly does not!) - MA will fallback to stop and resume. So if you hit pause, it will stop the stream, save the state and next time you hit play, it will resume where you left off.

If that is not working correctly, there is an issue in the snapcast provider in MA. 
I don't use snapcast at all so I do not test/know this but if the snapcast provider is providing accurate timestamp information of the playback, resuming should work just fine. 

Basically as a user you shouldnt even notice the difference between stop+resume and actual pause. Maybe a 200ms delay, totally  neglegatble.

**Comment by SantiagoSotoC** [comment]
The queue problem should already be fixed in the latest betas, I don't know if it is ported to the stable version

**Reply by PureTryOut** [reply]
The queue problem as in pausing resets the queue back to previous songs rather than just pausing? That's my main problem (I just realized the issue I linked describes a different problem which I am indeed also having but unrelated) so I would be glad if it is, but like I said I tried 2.3.0b29 and it still occurred.

**Reply by SantiagoSotoC** [reply]
Let me try, I'm sure this was fixed, but if the bug is back, let's see why.

**Reply by SantiagoSotoC** [reply]
@marcelveldt The problem with the playback queue is back.
Before it fails one song back but now it comes back several songs

**Reply by PureTryOut** [reply]
Exactly :sweat_smile:  If it was just one song it wouldn't be so back but now it can go an hour or more worth of songs back if you listened for that long.

**Reply by marcelveldt** [reply]
You have to be more specific here folks - "the problem with the playback queue" is a super vague description.

The only global issue I'm aware of is that sometimes some songs are skipped in the queue or it stops playing.

**Reply by PureTryOut** [reply]
I think it's pretty clearly described in this entire ~~issue~~discussion :thinking: 

When you play a few songs and then pause, the queue is set back to the original point where you started playing. That means you go back in history _several_ songs and this can go up to hours if you've been playing without pausing that long.

**Reply by OzGav** [reply]
Have the latest releases fixed this?

**Reply by PureTryOut** [reply]
Nope, still happening as of 2.3.1

**Reply by OzGav** [reply]
Do you have any players other than snapcast to try?

**Reply by SantiagoSotoC** [reply]
I just made a video, at x60 speed showing the error. 

https://github.com/user-attachments/assets/b25f87f8-674c-4e46-b3c3-61767ca5d792

**Reply by SantiagoSotoC** [reply]
It is without audio but the ui shows quite accurately the status of the audio, before pausing it, after resuming the pause, it goes back to the first song in the queue, no matter how many songs are played.

**Reply by SantiagoSotoC** [reply]
Well I think I found the problem, @marcelveldt  does it make sense that I have to set player.elapsed_time = time.time() when there is a stop ?

**Reply by SantiagoSotoC** [reply, marked as answer]
https://github.com/music-assistant/server/pull/1764

**Comment by cyber553** [comment]
I am having a similar issue, except with a few different component in my setup, but I'm also using the latest version of MA (2.5.8) on HAOS 16.2, Core 2025.9.4, and frontend 20250903.5.

I have Sonos speaker connected to one Home Assistant device on my Media VLAN. This is connected to my main Home Assistant device on my main VLAN via Remote Home-Assistant (HACS integration), which provides media_player entities for each of my Sonos players that mirror those created by the Sonos Integration.

I have added these to MA using the Home Assistant Plug-in and Home Assistant MediaPlayers Player Provider. The Sonos media players all play music from my NAS server well and the devices group/ungroup/sync nicely in MA on my main HA device.

The issue I'm experiencing is that when I play a playlist of songs from my NAS, If I don't manually skip tracks forward and back, the MA UI will constantly show the song that I most recently manually skipped to, but won't advance to the next song, even though the audio coming from the speaker does properly advance to the next song in the playlist, as desired. When I press stop/pause, then play again, when the music resumes, it will always go back to the beginning of the last song that I manually skipped it to playing in the playlist, not where it was when I pressed pause/stop seconds ago.

I love the power of Music Assistant and it does so many things incredibly well. I'm just struggling with the annoyance of this issue with basic functionality. Any ideas for fixes would be greatly appreciated!

**Reply by marcelveldt** [reply]
For Sonos speakers, use the Sonos provider in MA. Only use the HA provider as a last resort if there is no native option

**Reply by cyber553** [reply]
I did try that, but unfortunately, the Sonos integration doesn't work across VLANs, HA has to be on the same VLAN as the Sonos speakers for it to work, and my speakers are on an untrusted VLAN, while my HA needs to be on a trusted VLAN.

This is why I am using a second HA device, on the same VLAN as the speakers, to pull the devices/entities into HA (using the HA Sonos integration), and then using the Remote Home-Assistant (HACS integration) to pull entities (e.g. media_player.*) from that instance into my main HA device on a secure/trusted VLAN.

This seems to work as intended for everything in MA except for the issue I've describe above. But it does require me to use the Home Assistant Plug-in and HA MediaPlayers Provider in MA.

**Reply by OzGav** [reply]
We recommend you put everything on the same network and use the MA providers. We don’t support playing across VLANs. Sorry.

**Reply by cyber553** [reply]
While I am playing across VLANs (I don’t trust Sonos’ security enough to put those devices on a trusted VLAN), I have that part working properly using the Remote Home-Assistant integration. I am using the “Home Assistant” Plugin Provider and “Home Assistant MediaPlayers” Player Provider in MA to pull in the media_player entities so MA can use them. 

As far as I can tell, these providers are all built into MA. Everything with my setup seems to work properly in MA, with the only exception being this funky track playback behavior.
